Update readme with build steps (#120)
concord-consortium/CEASAR: README.md

In-Editor testing: The network panel is set to show extra entries in Editor mode - this proved useful in UI design, and is a useful feature (using conditional compilation) should this be required elsewhere. Much of the `SceneLoader` component uses conditional compilation to set up cameras and UI for different build targets.

### Unity Cloud Build

> As of July, 2021 Unity Cloud Build are no longer running. Previously Cloud Builds were created from the `master` branch for Mac and Windows. Instead builds are done locally and then shared via S3. Furthermore, Cloud Builds were not much use beyond Mac and Windows build targets, since Unity Cloud Builds do not support the Quest or Hololens.

Reviewer: ```suggestion
As of July 2021 Unity Cloud Build is no longer running. Previously Cloud Builds were created from the `master` branch for Mac and Windows. Instead builds are done locally and then shared via S3. Furthermore, Cloud Builds were not much use beyond Mac and Windows build targets, since Unity Cloud Builds do not support the Quest or Hololens.
```
or perhaps "Unity Cloud Builds are no longer running."

### Building for WebGL
To build for `WebGL`:
* in the Unity editor, change the build target to `WebGL`
* Rename the `Oculus` folder to `Oculus~` to hide the folder from Unity
* Do the build to WebGL
* When the build has completed, Rename the `Oculus~` folder to `Oculus` to return the folder to normal

> * By default, the Unity WebGL build uses static 960 x 600 pixel dimensions for the content frame. CSS adjustments can be made to the `index.html` file to change how the CEASAR content frame is shown in the browser tab.

Reviewer: ```suggestion
* By default, the Unity WebGL build uses static 960 x 600 pixel dimensions for the content frame. CSS adjustments can be made to the `index.html` file to change how the CEASAR content frame is shown in the browser.
```

While this build process may work on Mac (untested) on the PC you gain the ability to use the Oculus Quest in `Link` mode for live in-editor debugging while wearing the headset, if your headset is set up correctly (Developer mode needs to be enabled on the device, which can only be done via the Oculus iOS / Android application for administering your device).

### Building for Hololens2

> Make sure you are using the `hl2-2019` branch when working with the HoloLens. Full information on Hololens build configuration consult [the MRTK installation guide](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Installation.html).

Reviewer: ```suggestion
Make sure you are using the `hl2-2019` branch when working with the HoloLens. For full information on Hololens build configuration consult [the MRTK installation guide](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Installation.html).
```

#### Prerequisites

> The following prerequisites are required to build and deploy to a HoloLens.

Reviewer: ```suggestion
The following prerequisites are required to build and deploy to a HoloLens as of July 2021.
```
Since there are dates embedded in the versions that may change.

- Install Unity 2019.4.18f1.
- Install the Unity 2019.4 Universal Windows Platform module.
- Install Visual Studio 2019.
- Using the Visual Studio Installer, install the Visual Studio Universal Platform development Workload.
- Using the Visual Studio Installer, install the Visual Studio Game development with Unity Workload.
- Using the Visual Studio Installer, install the Visual Studio Universal Platform development Workload.
- Using the Visual Studio Installer, install the USB Device Connectivity Individual Component.

> - Using the Visual Studio Installer, ensure that the Windows 10 SDK Individual Component is installed (this should be installed with the Universal Platform development Workload).

Reviewer: You could eliminate the redundancy with:

- Using the Visual Studio Installer:
    - install the Visual Studio Universal Platform development Workload.
    - install the Visual Studio Game development with Unity Workload.
    - install the Visual Studio Universal Platform development Workload.
    - install the USB Device Connectivity Individual Component.
    - ensure that the Windows 10 SDK Individual Component is installed (this should be installed with the Universal Platform development Workload).

#### Unity Settings
Unity is used to create a Visual Studio solution from the Unity CEASAR project. Ensure that the following settings and configurations are enabled before building.

> - Ensure that Unity is pointing at your current installation of Visual Studio. From the Unity menu select Edit > Preferences. Set the installed version of Visual Studio 2019 as the External Script Editor (you may need to browse to the installed Visual Studio executable in your Program Files folder).

Reviewer: - Ensure that Unity is pointing at your current installation of Visual Studio.
    - From the Unity menu select Edit > Preferences.
    - Set the installed version of Visual Studio 2019 as the External Script Editor (you may need to browse to the installed Visual Studio executable in your Program Files folder).

- Turn off Copy PDB Files (turn this on if you plan to debug using either of the methods above).

#### Build Visual Studio Solution in Unity
After setting up all of the required prerequisites and configuring Unity, you can build the Visual Studio solution.
- In the Unity Build Settings, press Build and select a location for Unity to build your project. Unity does not build a finished HoloLens application. Instead it builds a .sln solution file (and all required supporting files) that can be opened in Visual Studio.
- Once the build is complete, open the solution in Visual Studio (double-click the .sln file).

#### Build and Deploy to HoloLens
With the Visual Studio solution, we can build and deploy the CEASAR application to the HoloLens.
- To connect a PC to a HoloLens, you must first enable Developer Mode on the PC. This step is only required once. Open Settings on your PC. Select Update and Security. Select For developers. Enable Developer Mode and select Yes to accept the change (this might require downloading a Windows update before this step is complete).

> - Connect the HoloLens to your PC using USB-C to USB-A cable.

Reviewer: ```suggestion
- Connect the HoloLens to your PC (using USB-C to USB-A cable if necessary).
```

- Once the deployment is complete, CEASAR will appear in your list of HoloLens applications and can be run on your HoloLens.
- The first time that you connect the HoloLens to your PC and attempt to deploy to it, you will need to use a Pairing Pin. In the HoloLens, select Settings > Update > For developers and enable Developer Mode. Enable the Device Portal. Select Pair and the HoloLens will generate a Pairing Pin that can be entered in Visual Studio. Before selecting the HoloLens device from the menu, you will need to enter the Pairing Pin.

#### Remote Debugging
You can use remote debugging in Visual Studio to connect to an installed CEASAR application running on the HoloLens and see errors, exceptions, and debug statements in the Visual Studio Output window.
- Open the CEASAR solution that you built using Unity in Visual Studio (double-click the .sln file).
- Set the Solution Configuration to Debug.
- Build and deploy CEASAR to your Hololens using the above steps.
- From the Debug menu in Visual Studio, select Other Debug Targets > Debug Installed App Package.
- Set the Connection Type to Device.
- Select CEASAR from the list of App Packages and press Start.

> - The CEASAR application will run on your HoloLens. As you use the application, Unity debug statements will appear in the Output window. The framerate will be significantly worse in this debug mode.

Reviewer: ```suggestion
- The CEASAR application will run on your HoloLens. As you use the application, Unity debug statements will appear in the Output window. The frame rate will be significantly lower in this debug mode.
```
